cmd/aro/dbtoken.go

```diff
@@ -1,130 +0,0 @@
-package main
-
-// Copyright (c) Microsoft Corporation.
-// Licensed under the Apache License 2.0.
-
-import (
-	"context"
-	"net"
-	"os"
-
-	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm/policy"
-	"github.com/sirupsen/logrus"
-
-	"github.com/Azure/ARO-RP/pkg/database"
-	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
-	pkgdbtoken "github.com/Azure/ARO-RP/pkg/dbtoken"
-	"github.com/Azure/ARO-RP/pkg/env"
-	"github.com/Azure/ARO-RP/pkg/metrics/statsd"
-	"github.com/Azure/ARO-RP/pkg/metrics/statsd/golang"
-	"github.com/Azure/ARO-RP/pkg/util/keyvault"
-	"github.com/Azure/ARO-RP/pkg/util/oidc"
-)
-
-func dbtoken(ctx context.Context, log *logrus.Entry) error {
-	_env, err := env.NewCore(ctx, log, env.COMPONENT_DBTOKEN)
-	if err != nil {
-		return err
-	}
-
-	if err := env.ValidateVars("AZURE_GATEWAY_SERVICE_PRINCIPAL_ID", "AZURE_DBTOKEN_CLIENT_ID"); err != nil {
-		return err
-	}
-
-	if !_env.IsLocalDevelopmentMode() {
-		if err := env.ValidateVars("MDM_ACCOUNT", "MDM_NAMESPACE"); err != nil {
-			return err
-		}
-	}
-
-	msiToken, err := _env.NewMSITokenCredential()
-	if err != nil {
-		return err
-	}
-
-	msiKVAuthorizer, err := _env.NewMSIAuthorizer(_env.Environment().KeyVaultScope)
-	if err != nil {
-		return err
-	}
-
-	m := statsd.New(ctx, log.WithField("component", "dbtoken"), _env, os.Getenv("MDM_ACCOUNT"), os.Getenv("MDM_NAMESPACE"), os.Getenv("MDM_STATSD_SOCKET"))
-
-	g, err := golang.NewMetrics(log.WithField("component", "dbtoken"), m)
-	if err != nil {
-		return err
-	}
-
-	go g.Run()
-
-	if err := env.ValidateVars(envDatabaseAccountName); err != nil {
-		return err
-	}
-
-	dbAccountName := os.Getenv(envDatabaseAccountName)
-
-	clientOptions := &policy.ClientOptions{
-		ClientOptions: _env.Environment().ManagedIdentityCredentialOptions().ClientOptions,
-	}
-	logrusEntry := log.WithField("component", "database")
-	dbAuthorizer, err := database.NewMasterKeyAuthorizer(ctx, logrusEntry, msiToken, clientOptions, _env.SubscriptionID(), _env.ResourceGroup(), dbAccountName)
-	if err != nil {
-		return err
-	}
-
-	dbc, err := database.NewDatabaseClient(log.WithField("component", "database"), _env, dbAuthorizer, m, nil, dbAccountName)
-	if err != nil {
-		return err
-	}
-
-	dbName, err := DBName(_env.IsLocalDevelopmentMode())
-	if err != nil {
-		return err
-	}
-
-	userc := cosmosdb.NewUserClient(dbc, dbName)
-
-	err = pkgdbtoken.ConfigurePermissions(ctx, dbName, userc)
```

Comment:
The original dbtoken only gains permissions over the cluster database, not the entire database account, if i'm understanding the original purpose of dbtoken originally. We are now using an MSI credential which suggests that we won't be scoping our permissions down to the cluster database anymore, but instead granting permissions over the whole database account. Is this a correct understanding, or am I misunderstanding the scope?

Reply:
So these permission are for the data plane only not the account plane.
However I think I can narrow it down by specifying scope in the role assignment to be of the ARO database.

Reply:
Added the scope of the database rather than it being the database account. Please have a look.

Reply:
thank you!